### ISSUE TITLE
moves next element onto end of previous line

### ISSUE THREAD

**Issue by simonmcconnell** [surface-ui/surface_formatter#49]
Hi,

I thought I'd try with v0.6.0 but it is still moving stuff up onto the previous line.  I feel like it worked when I first tried it but lately it is shifting comments and other tags up onto the end of the line above it.  As you can see in the diff below, the commented out `div`, `GotoTime` and the start of the surface comment `{!--` all disappeared at the end of the `AlarmDivisionFilterDropdown`.  And for some reason it put the `<button ...` lines onto the one line.

![image](https://user-images.githubusercontent.com/22566656/138456069-8b452f70-779c-461d-926e-0a72b0019d59.png)

I'm on Elixir 1.12.3, OTP 24, surface & formatter 0.6 and Windows 10, though I use `LF` line endings.

**Comment by paulstatezny** [comment]
@simonmcconnell Thanks for your patience on this issue. Would you be willing to share the full contents of that `~F` sigil? If you don't want to do it publicly, you could put it in a private [gist](https://gist.github.com/) and send it to me (@paulstatezny) on Elixir Slack.

**Comment by paulstatezny** [comment]
@simonmcconnell I'm going to close this issue since it's been almost a month and I haven't heard back. But please feel free to reply back and I'm happy to re-open and investigate if this is still a problem on the latest version.

I'm puzzled by the behavior in your screenshot because it's contrary to the rules the formatter follows, and I've never seen it act like that, and nobody else has reported it.

It makes me wonder if it might be something like a separate editor plugin interfering.

**Comment by simonmcconnell** [comment]
@paulstatezny sorry for my tardiness.  It was definitely from running the mix task.  Anyhoo, I tried v`0.7.2` just now and it tidied things up nicely.  There was only one inconsistency that I noticed.

![image](https://user-images.githubusercontent.com/22566656/146622168-c202d81a-2941-48f1-8d4f-cd03208a9324.png)

Everywhere else it put the closing backet inline with the attribute.  So I manually edited it to the following to see how it would handle the `}` on its own line:
```
<div
  x-data={"{
    open: false,
    text: '#{@text}',
    originalText: '#{@text}'
  }"
  }
```

Then I ran the formatter and it made it even weirder, by indenting it all further:
```
<div
  x-data={"{
        open: false,
        text: '#{@text}',
        originalText: '#{@text}'
      }"}
```

Each time I run the formatter, it indents it even futher.  I have other `x-data` attributes that are plain text, i.e. don't have the HEEX `{}`s around them, which are unaffected.  It doesn't seem to like the string inside the squigglies being split over lines.

![image](https://user-images.githubusercontent.com/22566656/146622538-20c93c0a-aaaa-4c93-b33b-50e998d6a8ec.png)

**Comment by paulstatezny** [comment]
@simonmcconnell Thanks for reporting this. I'll take a look at it!

**Comment by paulstatezny** [comment]
@simonmcconnell And don't sweat the delayed response — I just appreciate the report!

**Comment by paulstatezny** [comment]
@simonmcconnell This was patched in #54 and the patch was released in v0.7.3

https://hex.pm/packages/surface_formatter/0.7.3

Thanks again for your valuable analysis and communication!
